hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V052AddressBookSchema.java

```diff
@@ -0,0 +1,104 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * General schema for the addressbook service
+ * (FUTURE) When mod-service release is finalized, rename this class to e.g.
+ * {@code Release47addressbookSchema} as it will no longer be appropriate to assume
+ * this schema is always correct for the current version of the software.
+ */
+public class V052AddressBookSchema extends Schema {
+    private static final Logger log = LogManager.getLogger(V052AddressBookSchema.class);
+
+    private static final long MAX_NODES = 100L;
+    private static final SemanticVersion VERSION =
+            SemanticVersion.newBuilder().major(0).minor(52).patch(0).build();
+
+    public V052AddressBookSchema() {
+        super(VERSION);
+    }
+
+    @NonNull
+    @Override
+    public Set<StateDefinition> statesToCreate() {
+        return Set.of(StateDefinition.onDisk(NODES_KEY, EntityNumber.PROTOBUF, Node.PROTOBUF, MAX_NODES));
+    }
+
+    @Override
+    public void migrate(@NonNull final MigrationContext ctx) {
+        requireNonNull(ctx);
+        final WritableKVState<EntityNumber, Node> writableNodes =
+                ctx.newStates().get(NODES_KEY);
+        final var networkInfo = ctx.networkInfo();
+        final var nodeConfig = ctx.configuration().getConfigData(NodesConfig.class);
+        final var addressBook = networkInfo.addressBook();
+        AtomicInteger totalNodes = new AtomicInteger(0);
+        log.info("Started migrating nodes from address book");
+
+        addressBook.forEach(nodeInfo -> {
+            final var node = Node.newBuilder()
+                    .nodeId(nodeInfo.nodeId())
+                    .accountId(nodeInfo.accountId())
+                    .description(nodeInfo.memo())
+                    .gossipEndpoint(List.of(
+                            ServiceEndpoint.newBuilder()
+                                    .ipAddressV4(Bytes.wrap(nodeInfo.internalHostName()))
+                                    .port(nodeInfo.internalPort())
+                                    .build(),
+                            ServiceEndpoint.newBuilder()
+                                    .ipAddressV4(Bytes.wrap(nodeInfo.externalHostName()))
+                                    .port(nodeInfo.externalPort())
+                                    .build()))
+                    .gossipCaCertificate(nodeInfo.sigCertBytes())
+                    .weight(nodeInfo.stake())
+                    .build();
+            writableNodes.put(
+                    EntityNumber.newBuilder().number(nodeInfo.nodeId()).build(), node);
+            totalNodes.getAndIncrement();
+        });
+        if (totalNodes.get() > nodeConfig.maxNumber()) {
+            log.error("Address book contains more nodes than the configured maximum of {}", nodeConfig.maxNumber());
```

Comment:
We shouldn't fail migrating imo. We can fail when we try to add nodes using the transactions.

Reply:
Removed failed migrating. Just log a warning message.
Thanks @Neeharika-Sompalli